opentok-whiteboard.js

```diff
@@ -94,95 +96,113 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
                     });
                 }
             };
-            
+
             scope.erase = function () {
-                //scope.color = element.css("background-color") || "#fff";
-                scope.lineWidth = 50;
                 scope.erasing = true;
             };
-            
+
             scope.capture = function () {
                 if (iOS) {
                     // On iOS you can put HTML in a mailto: link
-                    window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
+                    $window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
                 } else {
                     // We just open the image in a new window
-                    window.open(canvas.toDataURL('image/png'));
+                    $window.open(canvas.toDataURL('image/png'));
                 }
             };
 
             scope.undo = function () {
                 if (!undoStack.length)
                     return;
-                var undodata = undoStack.pop();
-                undoWhiteBoard(undodata);
-                redoStack.push(undodata);
-                sendUpdate('otWhiteboard_undo', undodata);
+                var id = undoStack.pop();
+                undoWhiteBoard(id);
+                sendUpdate('otWhiteboard_undo', id);
             };
 
-            var undoWhiteBoard = function (data) {
-                for (i = data.start - data.count; i < data.start; i++) {
-                    drawHistory[i].show = 0;
-                }
-                clearCanvas();
-                drawHistory.forEach(function (update) {
-                    draw(update);
+            var undoWhiteBoard = function (id) {
+                redoStack.push(id);
+                pathStack.some(function(path) {
```

Comment:
I used `some` because `forEach` doesn't have a built in way to break it and I want to break the loop once it finds the path id. There shouldn't be more than one path with the same id. Why do you think `forEach` would be better?